### ISSUE TITLE
stream_compression broken with --sudo

### ISSUE THREAD

**Issue by moben** [digint/btrbk#115]
with --sudo, `sudo` is prepended to `$SSH_ORIGINAL_COMMAND`, but when that is e.g. `gzip -d -c | btrfs receive /backup/`, only the `gzip` is run with sudo, so it fails with

    WARNING: [send/receive] (send=/data/backup/data/Data.20161119T1639_1, receive=/backup/data) At subvol /data/backup/data/Data.20161119T1639_1
    ERROR: cannot open destination directory /backup/data: Operation not permitted
    ERROR: Failed to send/receive btrfs subvolume: /data/backup/data/Data.20161119T1639_1 [/data/backup/data/Data.20161119T1639] -> <remote>:/backup/data

**Comment by digint** [comment]
I added (pretty much untested) support for `backend btrfs-progs-sudo` configuration option in the [sudo_backend branch](https://github.com/digint/btrbk/tree/sudo_backend), e6ded3a50fc21acd84531a6b588092d538621988. The ssh_filter_btrbk script is also adapted, make sure to set "--sudo" as first option.

If you only want sudo for remote hosts (which is probably your setup right now), set `backend_remote btrfs-progs-sudo`, and btrbk will only prefix remote (ssh) calls to "btrfs" with "sudo -n".

This was on my TODO list anyway, and probably needs some more detail work, but should work in principle.

Note that btrbk-0.24.0 comes with a backend "btrfs-progs-btrbk", which is still pretty experimental and requires more setup work, but ends in being more secure than using sudo.

**Comment by digint** [comment]
added `backend btrfs-progs-sudo` support and documentation in: cb82bd5fa43031de3e7344b12175f9a6f4332e01

(makes "sudo_backend" branch obsolete)

**Comment by digint** [comment]
@moben do you have an example /etc/sudoers file you could share? Best would be a snipped with distinct subcommands for:
```
btrfs subvolume list
btrfs subvolume show
btrfs subvolume snapshot
btrfs subvolume delete
btrfs send
btrfs receive
```

**Comment by digint** [comment]
`backend btrfs-progs-sudo` is now available in btrbk-v0.25.0, closing issue
